src/olympia/addons/tests/test_views.py

```diff
@@ -1975,6 +2077,59 @@ def test_metadata_required(self):
         response = self.client.patch(self.url, data=data)
         assert response.status_code == 200
 
+    def test_default_locale(self):
+        response = self.client.patch(self.url, data={'default_locale': 'zz'})
+        assert response.status_code == 400
+        assert response.data == {'default_locale': ['"zz" is not a valid choice.']}
+
+        Translation.objects.create(
+            id=self.addon.summary_id, locale='fr', localized_string='summary Française'
+        )
+        self.addon.description = 'description!'
+        self.addon.save()
+
+        # failure cases:
+        # A field is required and doesn't already have a value in new default
+        # B field is set to None in update
+        # C field isn't required, but has other translations already
+        # D field isn't required, but has other translations provided
+        response = self.client.patch(
+            self.url,
+            data={
+                'default_locale': 'fr',
+                # A no name, doesn't have a value in fr
+                'summary': {'fr': None},  # B summary has a value, but None would clear
+                # C no description, has a value in en-US already
+                'support_url': {'de': 'https://de.support.test/'},  # D
+            },
+        )
+        assert response.status_code == 400, response.data
+        error_string = 'A value in the default locale of "fr" is required.'
+        assert response.data == {
+            'name': [error_string],
+            'summary': [error_string],
+            'description': [error_string],
+            'support_url': [error_string],
+        }
+
+        self.addon.update(description=None)
+        # success cases:
```

Comment:
Ok, I see these success cases are tested here, but why mention them in the comments above if they are not being tested in those test cases?

Reply:
similarly to the addon create tests, they are being all tested at once.  The summary translation is created earlier in the test on line 2085; description is set to `None` just above

Reply:
I can add to the test comments?  What do you think would clarify what's going on?

Reply:
In this test it seems like they are all called out, whereas in the others they are not, so maybe you can do the same in the others, but ultimately I don't think it's a big deal. Do whatever you like (or nothing at all).